Dockerfile-cni-plugin

```diff
@@ -0,0 +1,40 @@
+# syntax=docker/dockerfile:1.4
+
+ARG BUILDPLATFORM=linux/amd64
```

Comment:
Does this have an effect on whether building happens strictly for `amd64`? i.e if we wanna build this for arm, will overriding this variable have an effect?  afaik the value is taken from the environment when docker is run.

Reply:
I think this was a mistake, I've taken `BUILDPLATFORM` out. An explicit goal here is to essentially duplicate what the `proxy-init` Dockerfile is doing and it's not doing this.